roles/openshift_builddefaults/tasks/main.yml

```diff
@@ -5,7 +5,9 @@
     # TODO: add ability to define builddefaults env vars sort of like this
     # may need to move the config generation to a filter however.
     # openshift_env: "{{ hostvars[inventory_hostname]
-    #                    | oo_merge_dicts(hostvars)
+    #                    | oo_merge_dicts(vars if (ansible_version.full
+    #                                              | version_compare('2.0', 'ge')
+    #                                              | bool) else hostvars)
```

Comment:
Would it be better to push the logic into a filter (say oo_merge_hostvars and pass it both hostvars and vars)?

It seems like it would be less error prone than doing the version compare multiple times in roles.